src/App.jsx

```diff
@@ -33,29 +37,41 @@ function App() {
   const onStart = () => {
     setStart(false);
     setReset(false);
+    setShowResult(false);
   };
 
   const onReset = () => {
     setReset(true);
     setStep(0);
     setCorrectAnswers(0);
     setStart(true);
+    setShowResult(false);
   };
 
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const res = await axios.get(API_URL);
+        const userToken = await axios.get(TOKEN);
+        const { token, response_code } = userToken.data;
+        setToken(token);
+        setResCode(response_code);
+        const res = await axios.get(`${API_URL}&token=${token}`);
         setData(res.data.results);
       } catch (error) {
-        notify();
+        if (resCode === 3) {
```

Comment:
It's unclear what's the purpose of the number `3` here.

If it's something given to you by the API, then make it a named constant.